src/charm.py

```diff
@@ -75,23 +74,24 @@ def __init__(self, *args):
     #  Charm Lifecycle Hooks
     # =======================
 
-    def _on_start(self, event: StartEvent) -> None:
-        """Renders basic PGB config.
-
-        Deferrals:
-            - If pgbouncer container cannot connect, since this makes rendering config impossible.
-            - If config is unavailable in filesystem or peer databag, and this unit isn't the
-              leader, and therefore can't generate a default config.
-            - When we try and update relation endpoints, and pebble throws an error, implying that
-              it hasn't started yet.
-        """
```

Comment:
I don't see the added benefit of having both `_on_start` and `_on_pgbouncer_pebble_ready` doing a part of the initialisation and waiting on each other. I moved most of the logic in `_on_start` in `_on_pgbouncer_pebble_ready` since there we should have the container running. Thoughts?